app/models/case_log.rb

```diff
@@ -25,6 +25,7 @@ class CaseLog < ApplicationRecord
   before_validation :reset_scheme_location!, if: :scheme_changed?, unless: :location_changed?
   before_validation :process_postcode_changes!, if: :postcode_full_changed?
   before_validation :process_previous_postcode_changes!, if: :ppostcode_full_changed?
+  before_validation :set_housingneeds_fields, if: :housingneeds_present?
```

Comment:
Is this really a special case thing or are these just derived values that could be in the derived variables?

Reply:
It was failing a test but it seems it was due to a set up of it